### PR TITLE
When sharing a link via e-mail set users e-mail as sender - take 2

### PR DESCRIPTION
**Pull request by rullzer**
New (trivial) fix for #10545.
Now setting the sender in the code that wants to send the e-mails.
This replaces #13853 

Again no unit tests as I have no idea how to mock this properly.

### CODE REVIEW

**Comment by PVince81** [comment]
@LukasReschke @nickvergessen

File: core/ajax/share.php [hunk 1]
```diff
@@ -105,7 +105,7 @@
 			// don't send a mail to the user who shared the file
 			$recipientList = array_diff($recipientList, array(\OCP\User::getUser()));
 
-			$mailNotification = new OC\Share\MailNotifications();
+			$mailNotification = new \OC\Share\MailNotifications(\OC::$server->getUserSession()->getUser()->getUID());
```
Comment (nickvergessen) [inline core/ajax/share.php]: `\OC::$server->getUserSession()->getUser()` could be null, although I don't think this code is reachable for non-users?

Comment (rullzer) [inline core/ajax/share.php]: That is correct, this code is only reachable for authenticated users.

... On an unrelated note. Why is it returning null and not throwing an exception?

**Comment by MorrisJobke** [comment]
@LukasReschke @nickvergessen Ping :)

**Comment by nickvergessen** [comment]
Well for me that breaks the emails, because my Mail hoster does not allow sending emails via smtp with an arbitrary "from":
```
The following From address failed: nickvergessen@owncloud.com : MAIL FROM command failed,550,Requested action not taken: mailbox unavailable
550 Sender address is not allowed.
```

So not sure whether that is what we want to do, if the mail settings are set:
```php
  'mail_from_address' => '***',
  'mail_domain' => '***',
```
Do we need another option to allow forcing it? :see_no_evil:

**Comment by LukasReschke** [comment]
No. That's not what the from field is from… Let me comment later on that.

**Comment by LukasReschke** [comment]
https://github.com/owncloud/core/pull/15541 is an alternative approach that uses the `Reply-To` field. This should work in all cases regardless of SPF restrictions etc.

**Comment by LukasReschke** [comment]
Thing is that we can't just use the users' address as `Sender` or `From` field, there are quite some cases where this will explode and thus it's easier to go this route :-)

**Comment by rullzer** [comment]
Yep that sounds like way less work. Lets close this one and continue the discussion there.

**Pull request closed without merging**